test/scripts/gh-actions/setup-deps.sh

```diff
@@ -16,51 +16,72 @@
 
 # The script will install KServe dependencies in the GH Actions environment.
 # (Istio, Knative, cert-manager, kustomize, yq)
+# Usage: setup-deps.sh $DEPLOYMENT_MODE $NETWORK_LAYER
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]:-$0}")" &>/dev/null && pwd 2>/dev/null)"
 DEPLOYMENT_MODE="${1:-'serverless'}"
+NETWORK_LAYER="${2:-'istio'}"
 
-ISTIO_VERSION="1.20.4"
-CERT_MANAGER_VERSION="v1.15.1"
+ISTIO_VERSION="1.23.2"
+CERT_MANAGER_VERSION="v1.16.1"
 YQ_VERSION="v4.28.1"
+GATEWAY_API_VERSION="v1.2.1"
+ENVOY_GATEWAY_VERSION="v1.2.2"
```

Comment:
Can you clarify if these versions are the minimum versions required? Are there requirements from these versions for others besides GW API?

Reply:
Oh this is to be consistent with the current install script https://github.com/kserve/kserve/pull/3952/files#diff-f2775414d7207d0cdb298bdfcb622a135d58ca38dcbd44dd4dbe4e51243fc7a9R20-R24